### PR TITLE
fix: copyAuthURL breaking Snyk in Sublime [ROAD-1166]

### PR DESCRIPTION
**Pull request by j-luong**
### Description

_This PR should fix scans not working in sublime due to an errant `break` in the `copyAuthURL` logic._

### Checklist

- [x] Tests added and all succeed
- [x] Linted

### CODE REVIEW

File: infrastructure/cli/auth/provider.go [hunk 1]
```diff
@@ -75,35 +77,29 @@ func (a *CliAuthenticationProvider) authenticate(ctx context.Context) error {
...
+		scanner := bufio.NewScanner(reader)
+		for scanner.Scan() {
+			text := scanner.Text()
+			url := a.getAuthURL(text)
+			out.Write(scanner.Bytes())
+			log.Debug().Str("method", "authenticate").Msgf("current auth url line: %s", text)
+
+			if url != "" {
+				a.authURL = url
+				log.Debug().Str("method", "authenticate").Msgf("found URL: %s", url)
```
[the remark this answers is not in the document]
Comment (j-luong) [reply infrastructure/cli/auth/provider.go:93]: Yeah, it seems like it. Removed the `break` and the code scan worked.

**Comment by michelkaporin** [comment]
@j-luong please link JIRA ticket in the PR title. Also what problem **exactly** does it lead to, when `break` termination is present? Is it hanging in authentication process?

File: infrastructure/cli/auth/provider.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (michelkaporin) [reply infrastructure/cli/auth/provider.go:92]: @j-luong can you please explain how break impacts the code flow in this case?

[the remark this answers is not in the document]
Comment (j-luong) [reply infrastructure/cli/auth/provider.go:92]: So it looks like the `break` exits the the PipeReader prematurely. The [docs](https://pkg.go.dev/io#Pipe) mention that each write to the PipeWriter blocks until it has satisfied one or more reads from the PipeReader. As we exit the PipeReader, not reading the entire pipe will cause the call to `runCliCmd` to never finish.

**Review by bastiandoetsch — APPROVED** [review]
LGTM

**Pull request merged**